extension-style/src/main/java/com/mapbox/maps/extension/style/layers/LayerExt.kt

```diff
@@ -18,24 +19,26 @@ import com.mapbox.maps.logE
  * @return StyleLayerPlugin
  */
 fun StyleManagerInterface.getLayer(layerId: String): Layer? {
-  return this.getStyleLayerProperty(layerId, "type").silentUnwrap<String>()?.let { type ->
-    when (type) {
-      "background" -> BackgroundLayer(layerId).also { it.delegate = this }
-      "location-indicator" -> LocationIndicatorLayer(layerId).also { it.delegate = this }
-      "sky" -> SkyLayer(layerId).also { it.delegate = this }
-      "circle" -> CircleLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "fill-extrusion" -> FillExtrusionLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "fill" -> FillLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "heatmap" -> HeatmapLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "hillshade" -> HillshadeLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "line" -> LineLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "raster" -> RasterLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      "symbol" -> SymbolLayer(layerId, this.getStyleLayerProperty(layerId, "source").unwrap()).also { it.delegate = this }
-      else -> {
-        logE(TAG, "Layer type: $type unknown.")
-        null
-      }
+  val source by lazy { getStyleLayerProperty(layerId, "source").unwrap<String>() }
```

Comment:
There're layers which do not have a "source" property, and that's why we specifically call this function inside the layer constructors.
i.e. background/location-indicator/sky layer doesn't have/support "source" property, and will report error here if you unwrap it to a String.

Reply:
I used lazy evaluation for `source` so it should not be evaluated until the variable is actually used, so I think it should be safe to do this since source is not used in the BackgroundLayer, LocationIndicatorLayer, SkyLayer constructors.

Reply:
ah, yes, sorry I missed lazy evaluation, sounds good!